src/dbus.cpp

```diff
@@ -0,0 +1,146 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2021 YADRO
+
+#include "dbus.hpp"
+
+#include <phosphor-logging/log.hpp>
+#include <xyz/openbmc_project/Common/error.hpp>
+
+using namespace phosphor::logging;
+using namespace sdbusplus::xyz::openbmc_project::Common::Error;
+
+/**
+ * @brief Make variable key.
+ *
+ * @param[in] name Variable name
+ * @param[in] guid Vendor GUID
+ *
+ * @return variable key
+ *
+ * @throw InvalidArgument in case of errors
+ */
+static VariableKey makeKey(const std::string& name,
+                           const std::vector<uint8_t>& guid)
+{
+    if (guid.size() != sizeof(uuid_t))
+    {
+        throw InvalidArgument();
+    }
+    VariableKey key;
+    key.name = name;
+    std::copy(guid.begin(), guid.end(), key.guid);
+    return key;
+}
+
+DBus::DBus(sdbusplus::bus::bus& bus, Storage& varStorage) :
+    Super(bus, objectPath), storage(varStorage)
+{}
+
+std::tuple<uint32_t, std::vector<uint8_t>>
+    DBus::getVariable(std::string name, std::vector<uint8_t> guid)
+{
+    if (storage.empty())
+    {
+        throw NotAllowed(); // it should be "Unavailable", but we have too old
+                            // DBus interfaces in the Vegman repo
+    }
+    const VariableKey key = makeKey(name, guid);
+    auto variable = storage.get(key);
+    if (!variable)
+    {
+        throw ResourceNotFound();
+    }
+    return std::make_tuple(variable->attributes, variable->data);
+}
+
+void DBus::setVariable(std::string name, std::vector<uint8_t> guid,
+                       uint32_t attributes, std::vector<uint8_t> data)
+{
+    const VariableKey key = makeKey(name, guid);
+    try
+    {
+        storage.set(key, VariableValue{attributes, data});
+    }
+    catch (const std::exception& ex)
+    {
+        log<level::ERR>("Error processing SetVariable method",
+                        entry("EXCEPTION=%s", ex.what()));
+        throw sdbusplus::xyz::openbmc_project::Common::Error::InternalFailure();
```

Comment:
The most upstream projects use it as:
```cpp
elog<InternalFailure>();
```

Reply:
But elog doesn't throw an exception, does it?
I don't see a reason to add second log record.

Reply:
Yes, `elog` adds the second log record with the exception name and then [throws it](https://github.com/openbmc/phosphor-logging/blob/ef952af247f12a0b77a06bbfe7813b268b615f4d/lib/include/phosphor-logging/elog.hpp#L160].)
I guess, these messages might be used by other services to create SEL for example.
But for this project, it's not actual, as it's for VEGMAN only. 
So, I don't insist on fixing.